pkg/controller/controller_test.go

```diff
@@ -630,6 +666,7 @@ func (s *ControllerSuite) TestExecActionSet(c *C) {
 			if !cancel {
 				err = s.waitOnActionSetState(c, as, final)
 				c.Assert(err, IsNil, Commentf("Failed case: %s", tc.name))
+				c.Assert(getCounterVecValue(s.ctrl.metrics.actionSetResolutionCounterVec, []string{tc.metricResolution}), Equals, oldValue+1, Commentf("Failed case: %s", tc.name))
```

Comment:
What happens in the cancel case?

Reply:
We're currently maintaining only `success` and `failure` resolutions as per our discussions. Hence `cancel` has essentially been ignored, and we will not be incrementing any values for the cancel case. 

If we do decide to add `cancel` in the future, then we will have to add a new `Inc()` at the point where we figure out that the `ActionSet` has been cancelled.

Reply:
It is a bit confusing because `TestExecActionSet` has been set up to "expect" a `failure` in the cancel case, but we aren't asserting anything for now. Let me know if it's better to pass in blank resolution status during the unit test.